registry/reference.go

```diff
@@ -164,13 +175,26 @@ func (r Reference) ValidateReference() error {
 	if r.Reference == "" {
 		return nil
 	}
-	if _, err := r.Digest(); err == nil {
+
+	return fmt.Errorf("%w: invalid non-tag, non-digest reference: %v", errdef.ErrInvalidReference, r.Reference)
+}
+
+// ValidateReferenceAsTag validates the reference as a tag.
+func (r Reference) ValidateReferenceAsTag() error {
+	if tagRegexp.MatchString(r.Reference) {
 		return nil
 	}
-	if !tagRegexp.MatchString(r.Reference) {
-		return fmt.Errorf("%w: invalid tag", errdef.ErrInvalidReference)
+
+	return fmt.Errorf("%w: invalid tag", errdef.ErrInvalidReference)
```

Comment:
It is more readable and maintainable to have code blocks

```go
if err := doStuff(); err != nil {
    // handle error
}
return nil
```

over

```go
if err := doStuff(); err == nil {
    return nil
}

// handle error
```

Reply:
I didn't know that, I'll make the change.